docs/src/grids.md

```diff
@@ -0,0 +1,684 @@
+function latitude_faces(j)
+    if j == 1 # equator
+        return 0
+    else # crudely estimate the location of the jth face 
+        φ₋ = latitude_faces(j-1)
+        φ′ = φ₋ + m * scale_factor(φ₋) / 2
+        return φ₋ + m * scale_factor(φ′)
+    end
+end    
+
+Lx = 360
+Nx = Int(Lx / m)
+
+# Deduce number of cells south of 90ᵒN
+λf = latitude_faces.(1:Nx)
+Ny = findfirst(λf .> 90) - 2
+
+grid = LatitudeLongitudeGrid(size = (Nx, Ny),
+                             longitude = (0, Lx),
+                             latitude = latitude_faces,
+                             topology = (Bounded, Bounded, Flat))
+```
+
+```@example plot
+φ = φnodes(grid, Center())
+Δx = xspacings(grid, Center(), Center(), with_halos=true)[1:Ny]
+Δy = yspacings(grid, Center())[1:Ny]
+
+using CairoMakie
+CairoMakie.activate!(type = "svg") # hide
+
+fig = Figure(size=(800, 400))
+axx = Axis(fig[1, 1], xlabel="Zonal spacing (km)", ylabel="Latitude (degrees)")
+scatter!(axx, Δx ./ 1e3, φ)
+
+axy = Axis(fig[1, 2], xlabel="Meridional spacing (km)")
+scatter!(axy, Δy ./ 1e3, φ)
+
+hidespines!(axx, :t, :r)
+hidespines!(axy, :t, :l, :r)
+hideydecorations!(axy, grid=false)
+
+display(fig); save("plot_lat_lon_mercator.svg", fig); nothing # hide
+```
+
+![](plot_lat_lon_mercator.svg)
+
+## Single-precision `RectilinearGrid`
+
+To build a grid whose fields are represented with single-precision floating point values,
+we specify the `float_type` along with the (optional) `architecture`,
+
+```jldoctest grids
+architecture = CPU()
+float_type = Float32
+
+grid = RectilinearGrid(architecture, float_type,
+                       topology = (Periodic, Periodic, Bounded),
+                       size = (16, 8, 4),
+                       x = (0, 64),
+                       y = (0, 32),
+                       z = (0, 8))
+
+# output
+16×8×4 RectilinearGrid{Float32, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── Periodic x ∈ [0.0, 64.0) regularly spaced with Δx=4.0
+├── Periodic y ∈ [0.0, 32.0) regularly spaced with Δy=4.0
+└── Bounded  z ∈ [0.0, 8.0]  regularly spaced with Δz=2.0
+```
+
+Single precision should be used with care.
+Users interested in performing single-precision simulations should get in touch via
+[Discussions](https://github.com/CliMA/Oceananigans.jl/discussions),
+and should subject their work to extensive testing and validation.
+
+For more examples see [`RectilinearGrid`](@ref) and [`LatitudeLongitudeGrid`](@ref).
+
+## Distributed grids
+
+To run the next examples, we have to launch julia with MPI and things are a bit more involved.
+For best results, create [a new environment](https://pkgdocs.julialang.org/v1/environments/)
+in an empty folder, and then add both `Oceananigans.jl` and `MPI.jl` to the new environment.
+Pasting this snipped into the terminal should do the trick:
+
+```bash
+mkdir OceananigansDistributedExamples
+cd OceananigansDistributedExamples
+touch Project.toml
+julia --project -e 'using Pkg; Pkg.add(["Oceananigans", "MPI"])'
```

Comment:
`touch` is unix command; this will work in more machines:

```suggestion
julia --project -e 'using Pkg; using Pkg; Pkg.activate("."); Pkg.add(["Oceananigans", "MPI"])'
```

Reply:
I think not